src/TwigComponent/src/ComponentFactory.php

```diff
@@ -43,7 +45,14 @@ public function metadataFor(string $name): ComponentMetadata
         $name = $this->classMap[$name] ?? $name;
 
         if (!$config = $this->config[$name] ?? null) {
-            $this->throwUnknownComponentException($name);
+            if (($template = $this->findAnonymousComponentTemplate($name)) !== null) {
+                return new ComponentMetadata([
+                    'key' => $name,
+                    'template' => $template,
+                ]);
+            }
+
+            throw new \InvalidArgumentException(sprintf('Unknown component "%s". The registered components are: %s. And no template anonymous component founded', $name, implode(', ', array_keys($this->config))));
```

Comment:
If you have some idea to make this error more helpful with the anonymous component...

Reply:
```suggestion
            throw new \InvalidArgumentException(sprintf('Unknown component "%s". The registered components are: %s. And no matching anonymous component template was found.', $name, implode(', ', array_keys($this->config))));
```

And we can move it into `throwUnknownComponentException()`, I think, and call it from both places.